### PR TITLE
argo: 2.2.1 -> 2.4.1

### PR DESCRIPTION
**Pull request by groodt**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

**Comment by groodt** [comment]
@GrahamcOfBorg build argo

**Comment by jonringer** [comment]
@GrahamcOfBorg build argo

**Comment by jonringer** [comment]
you need to be whitelisted for ofborg to run your commands :)

**Review by jonringer — APPROVED** [review]
`nix-review` passes on NixOS
diff LGTM
executable seems to work

```
[63 built, 2 copied (4.8 MiB), 0.7 MiB DL]
https://github.com/NixOS/nixpkgs/pull/70672
1 package were build:
argo
```

cc @kalbasit for go input

File: pkgs/applications/networking/cluster/argo/default.nix [hunk 1]
```diff
@@ -2,17 +2,18 @@
...
     owner = "argoproj";
     repo = "argo";
     rev = "v${version}";
-    sha256 = "0x3aizwbqkg2712021wcq4chmwjhw2df702wbr6zd2a2cdypwb67";
+    sha256 = "0idg0a21c0sq19202fybdwdypn6196qc7pyr9rrj3m08xq8vr6xm";
   };
 
   goDeps = ./deps.nix;
   goPackagePath = "github.com/argoproj/argo";
+  subPackages = [ "cmd/argo" ];
```
Comment (kalbasit) [inline pkgs/applications/networking/cluster/argo/default.nix]: Why switch to installing only `argo`? What about the other two commands?

Comment (groodt) [reply pkgs/applications/networking/cluster/argo/default.nix]: I can keep the other 2 binaries, but they are really only useful inside a Kubernetes Operator context when the full argo system is installed on a cluster.

The `argo` CLI is what I think is more relevant to users. It's more like a `argoctl`, but such a project does not exist.

I'm happy to add back the rest of the binaries if you prefer? Or I can add them back if somebody asks for them in the community?

Comment (kalbasit) [reply pkgs/applications/networking/cluster/argo/default.nix]: I'm ok to leave them off. We can add them if someone asks for them.

**Comment by kalbasit** [comment]
@GrahamcOfBorg build argo

**Review by kalbasit — APPROVED** [review]

File: pkgs/applications/networking/cluster/argo/default.nix (hunk 1 above)
Comment (jonringer) [reply pkgs/applications/networking/cluster/argo/default.nix]: another option would be to add a `argoFull` where they are available. But if the closure saving is minimal I'm okay with them being added back.

**Pull request merged**